Agent.py

```diff
@@ -0,0 +1,92 @@
+from Environment import *
+from copy import deepcopy
+import random
+import queue
+from itertools import combinations
+
+# from Tree import *
+
+
+class Agent(object):
+    def __init__(self, env, num_of_mine):
+        self.env = env
+        self.score = 0
+        self.kb = [[9 for col in range(self.env.dim)] for row in range(self.env.dim)]
+        self.num_cell_left = self.env.dim * self.env.dim
+        self.identified_hidden_cell = []
+        self.num_of_mine = num_of_mine
+
+        self.cell_to_inference = queue.Queue()
+        self.cell_unresolved = queue.Queue()
+
+    def inference(self):
+        # print(self.num_mine_found, len(self.identified_hidden_cell))
```

Comment:
here maybe we need to tell it when to stop by using if the number of cell left <0